### PR TITLE
Anyar/342 fix missing doc stats

### PR DESCRIPTION
**Pull request by aroltsch**
1. Fixes a bug around stats page. It uses `created_at` timestamp instead of `form8_started_at` timestamp to find `certifications` records. 
2. Add database_cleaner to tests

Testing:
1. Go to `/stats` page and test all tabs.
2. Reference this document: https://github.com/department-of-veterans-affairs/appeals-pm/blob/master/administration/requests/GAO%20100666.md#dashboard-metrics-for-caseflow-certification 
@astewarttistatech

### CODE REVIEW

**Comment by cmgiven** [comment]
👍 The calculations here track with what I'd expect. So glad we can use the existing data.

File: spec/rails_helper.rb [hunk 1]
```diff
@@ -80,11 +81,9 @@ class << base
 User.prepend(StubbableUser)
 
 def reset_application!
-  Task.delete_all
-  Appeal.delete_all
   User.clear_stub!
+  # database cleaner is set up to ignore users table so clean users here
```
Comment (joofsh) [inline spec/rails_helper.rb]: This looks great 👍 . Out of curiosity, why do we skip the users table? Did that mess up existing tests?

Comment (aroltsch) [inline spec/rails_helper.rb]: @joofsh It was due to a failing existing test. However, I figured it out the issue and will push in a few.

File: app/models/certification.rb [hunk 2]
```diff
@@ -56,37 +56,33 @@ def self.completed
...
+                   .or(was_missing_form9)
   end
 
   def self.was_missing_nod
-    where("nod_matching_at IS NULL or nod_matching_at > form8_started_at")
+    # Find certifications where 'nod_matching_at' is not set OR where NOD document was
+    # matched after the appeal was created in Caseflow.
+    # Allow 1 second lag just in case 'nod_matching_at' timestamp is a few milliseconds
+    # greater than 'created_at' timestamp
+    where(nod_matching_at: nil).or(where("nod_matching_at > created_at + INTERVAL '1 second'"))
```
Comment (aroltsch) [inline app/models/certification.rb]: @shanear I am not sure how to achieve the same in Ruby and also any Ruby code inside `where` turns into SQL.

File: app/models/certification.rb [hunk 3]
```diff
@@ -56,37 +56,33 @@ def self.completed
...
-          "or (form9_matching_at IS NULL or form9_matching_at > form8_started_at)")
+    was_missing_nod.or(was_missing_soc)
+                   .or(was_missing_ssoc)
+                   .or(was_missing_form9)
   end
 
   def self.was_missing_nod
-    where("nod_matching_at IS NULL or nod_matching_at > form8_started_at")
+    # Find certifications where 'nod_matching_at' is not set OR where NOD document was
+    # matched after the appeal was created in Caseflow.
```
Comment (aroltsch) [inline app/models/certification.rb]: Makes sense!

File: app/models/certification.rb (hunk 2 above)
Comment (shanear) [inline app/models/certification.rb]: @aroltsch you're right, I didn't read this statement correctly. All good 👍

File: spec/support/database_cleaner.rb [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+      # specs, so use truncation strategy.
+      DatabaseCleaner.strategy = :truncation
+    end
+  end
+
+  config.before(:each) do
+    DatabaseCleaner.start
+  end
+
+  config.append_after(:each) do
```
Comment (aroltsch) [inline spec/support/database_cleaner.rb]: It happens after each test. The time difference is in 10 seconds:
`real 3m41.295s` vs `real 3m31.587s`. I think we should have a clean DB state before each test is run. What do you think? @shanear

**Comment by cmgiven** [comment]
So this is what UAT is showing:

screen shot 2016-12-15 at 12 16 15 pm

Every case is shown as a case that had every document missing. There could be a good reason for this. Is date_added set?

**Pull request merged**